src/lib/features/dependent-features/dependent-features-store.ts

```diff
@@ -38,6 +38,23 @@ export class DependentFeaturesStore implements IDependentFeaturesStore {
         return rows.map((row) => row.child);
     }
 
+    async getParentOptions(child: string): Promise<string[]> {
+        const result = await this.db('features as f')
+            .where('f.name', child)
+            .select('f.project');
+        if (result.length === 0) {
+            return [];
+        }
+        const rows = await this.db('features as f')
+            .leftJoin('dependent_features as df', 'f.name', 'df.child')
+            .where('f.project', result[0].project)
+            .andWhere('f.name', '!=', child)
+            .andWhere('df.child', null)
+            .select('f.name');
+
+        return rows.map((item) => item.name);
```

Comment:
If the feature is already parent of child, should it still be returned?

Reply:
if the parent is already a child then it shouldn't (` .andWhere('df.child', null)`). If the parent is already a parent it's a non-issue.

Reply:
I am not fully sure when this query is being ran, but if child is feature A, if feature B is parent of A, should the query still return B?

Reply:
if feature B is not a child of any other parent then yes. If your parent is a child of any other feature it should never be returned.